woocommerce/woocommerce-ios
# [What's New] Support dark mode for What's New announcement icons

## Description


The `What's New Announcements` tool supports multiple icons. Previously, we always showed a single icon for both light and dark mode on the announcements screen, which stood out in dark mode.

This PR adds support for adaptively showing either a URL-based icon, or a base64 icon, in light or dark mode according to the user's preference.

For an announcement to support this feature, it should include two `icon` definitions for each feature, with `iconType: light` and `iconType: dark` as appropriate. These icons should have the same format as one another, i.e. be either a base64 icon by uploading a file, or a url-based icon by specifying the URL.

## Testing instructions


1. Update the version number in the project file to 15.9
2. Build and run the app, and sign in to the app using an a11n WordPress account
3. Observe that the What's New announcement is shown, and when you switch between dark and light mode, the images change accordingly.

Note that after you dismiss the What's New announcement, you can find it again under Menu > Settings > What's New in WooCommerce, as long as you're using a version for which there is an announcement.

This test covers both the `.adaptive` and `.adaptiveRemote` cases – the test announcement is set up with the Order-related icon as a URL image, and the Payment-related icon as a base64 image.

### Further testing
Update the announcement in the tool, found in Matticspace under Tools > Developers > Mobile Announce (ensure you edit the Woo iOS announcement targeting 15.9 only). Note that the announcement will have been stored in Core Data, and while you can go back and see it using the directions above, it will not be fetched from the network again. 

To see your changes, delete the app and run it again.

#### Suggested scenarios:
I've tested all of these, and it takes a little while, so feel free to take a selection or none of them.
- No `iconType` specified with a single URL icon – uses that icon in all modes
- No `iconType` specified with a single base64 icon – uses that icon in all modes
- `iconType: light` and `iconType: dark` specified with different icon formats – uses the light icon in all modes
- URL-based icons for one or both recognised iconTypes, with unreachable URLs – uses a placeholder
- Multiple icons specified with unrecognised `iconTypes` – uses the first icon in all modes
- Only `iconType: light` specified with a single URL icon – uses that icon in all modes
- Only `iconType: light` specified with a single base64 icon – uses that icon in all modes
- Only `iconType: "some other string"` specified with a single icon – uses that icon in all modes


## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/2472348/1bbf6e76-95f7-40cd-be61-53d07cd1f42d


---
